src/register/containers/register/register.component.ts

```diff
@@ -141,6 +141,15 @@ export class RegisterComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   public onPageContinue(formDraft: any): void {
+    if (formDraft && formDraft['controls']) {
+      if (Object.keys(formDraft.controls).filter(value => value.startsWith('PBANumber')).length > 1) {
+        Object.keys(formDraft.controls).filter(value => value.startsWith('PBANumber')).forEach(key => {
+          if (formDraft['controls'][key].value === null) {
+            this.store.dispatch(new fromStore.RemovePBANumber('remove' + key));
```

Comment:
there will be race condition on this.store.dispatch(new fromStore.RemovePBANumber('remove' + key));
and 
this.store.dispatch(new fromStore.SaveFormData({value, pageId: this.pageId}));

And it's not necessary to do two dispatches in one method, the best place to put this logic should be in 
this.store.dispatch(new fromStore.SaveFormData({value, pageId: this.pageId}));
and filter out those value is null/empty when saving them.

Reply:
@rayliangatos I was thinking of to implement this logic, in fact did it but issue is unless we save the data we get those PBA element which has no values which is not what expected.